system/jlib/jregexp.cpp

```diff
@@ -1344,10 +1344,11 @@ static bool WildMatchNreplace ( const char *src, int srclen, int srcidx,
                     (toupper(src[srcidx])!=toupper(next_char)))
                     goto Fail;
             }
-            else
+            else {
```

Comment:
I think you're correct.  I looked over the code and it looks like we want to increment the srcidx if the previous two conditionals are false.